js/game.js

```diff
@@ -0,0 +1,116 @@
+/** @type {HTMLCanvasElement} */
+
+/*GAME ENGINE */
+
+class Game {
+    constructor(ctx, width, height, player, canvas){
```

Comment:
You didn't need to pass this `canvas` property here on the constructor as you have the `ctx` already.